source/MaterialXTest/Runtime.cpp

```diff
@@ -1076,4 +1078,42 @@ TEST_CASE("Runtime: Traversal", "[runtime]")
     REQUIRE(bsdfCount == libBsdfCount);
 }
 
+TEST_CASE("Runtime: Looks", "[runtime]")
+{
+    mx::RtScopedApiHandle api;
+
+    mx::RtStagePtr stage = api->createStage(ROOT);
+
+    mx::RtPrim p1 = stage->createPrim("collection1", mx::RtCollection::typeName());
+    mx::RtCollection col1(p1);
+    mx::RtAttribute igeom = col1.getIncludeGeom();
+    igeom.setValueString("foo");
+    mx::RtAttribute egeom = col1.getExcludeGeom();
+    egeom.setValueString("bar");
+    REQUIRE(igeom.getValueString() == "foo");
+    REQUIRE(egeom.getValueString() == "bar");
+
+    mx::RtPrim p2 = stage->createPrim("child1", mx::RtCollection::typeName());
+    mx::RtPrim p3= stage->createPrim("child2", mx::RtCollection::typeName());
+    col1.addCollection(p2);
+    col1.addCollection(p3);
+    mx::RtRelationship rel = col1.getIncludeCollection();
+    REQUIRE(rel.targetCount() == 2); 
+    col1.removeCollection(p2);
+    REQUIRE(rel.targetCount() == 1);
+
+    mx::RtPrim pa = stage->createPrim("matassign1", mx::RtMaterialAssign::typeName());
+    mx::RtMaterialAssign assign1(pa);
+    assign1.getCollection().addTarget(p1);
+
+    //TODO : Add rest of testing..
+    mx::FileSearchPath searchPath(mx::FilePath::getCurrentPath() / mx::FilePath("libraries"));
+    api->setSearchPath(searchPath);
+    mx::DocumentPtr doc = mx::createDocument();
+    loadLibraries({ "stdlib", "pbrlib" }, searchPath, doc);
```

Comment:
The preferred way to handle library loading is now through the `RtApi` class. You already have a handle to the api instance here, so you can use:

```
api->setSearchPath(searchPath);
api->loadLibrary(STDLIB);
api->loadLibrary(PBRLIB);
```

And then all nodedefs are automatically registered, so in order to create node instances from a nodedef you just give the nodedef name as the _typename_ to `createPrim`. For example if there is a nodedef for a `surfacematerial` node you can use:

const mx::RtToken surfaceMaterialDef("ND_surfacematerial");
mx::RtNode material1 = stage->createPrim("/material1", surfaceMaterialDef);

Reply:
I've made the changes for this in the current code.